testsuite/features/step_definitions/api_common.rb

```diff
@@ -178,6 +178,36 @@
   assert_equal(false, ret.key?(attr))
 end
 
+Then(/^channel "([^"]*)" should (not )?be enabled on "([^"]*)"$/) do |channel, disabled, host|
```

Comment:
I'm not a big fan of grouping the enabled and the disabled case under one step.

The assert_equal test that results might be clever, but it's an horror to read and understand.

Besides, there's no performance benefit in factorizing.

Reply:
In opposite direction, I like it to use regex of that kind in our step definitions.
I found it easier to maintain aligned the code for the same kind of action.

Reply:
It's mostly to avoid having nearly identical code blocks with just a reversed check.
Beside that, there's no particular improvement as you say, so I have no strong preference on reverting it back if that's what we agree on.

Reply:
I like it, if I would complain in something about it is in the way the sentence is made, we have other examples with this kind of format:

```
When(/^I (enable|disable) repositories (before|after) installing branch server$/) do |action, when|
```